dot/network/service.go

```diff
@@ -414,47 +417,56 @@ func (s *Service) logPeerCount() {
 	}
 }
 
-func (s *Service) publishNetworkTelemetry(done <-chan struct{}) {
+func (s *Service) publishTelemetry(cancel <-chan struct{}) {
 	ticker := time.NewTicker(s.telemetryInterval)
 	defer ticker.Stop()
 
 	for {
+		s.sentBandwidthTelemetry()
+		err := s.sendBlockIntervalTelemetry()
+		if err != nil {
+			logger.Warnf("failed to sent block interval telemetry: %s", err)
+			continue
+		}
+
 		select {
-		case <-done:
+		case <-cancel:
 			return
-
 		case <-ticker.C:
-			o := s.host.bwc.GetBandwidthTotals()
-			s.telemetry.SendMessage(telemetry.NewBandwidth(o.RateIn, o.RateOut, s.host.peerCount()))
 		}
 	}
 }
 
-func (s *Service) sentBlockIntervalTelemetry() {
-	for {
-		best, err := s.blockState.BestBlockHeader()
-		if err != nil {
-			continue
-		}
-		bestHash := best.Hash()
+func (s *Service) sentBandwidthTelemetry() {
+	o := s.host.bwc.GetBandwidthTotals()
+	s.telemetry.SendMessage(telemetry.NewBandwidth(o.RateIn, o.RateOut, s.host.peerCount()))
+}
 
-		finalised, err := s.blockState.GetHighestFinalisedHeader()
-		if err != nil {
-			continue
-		}
-		finalizedHash := finalised.Hash()
+func (s *Service) sendBlockIntervalTelemetry() (err error) {
+	best, err := s.blockState.BestBlockHeader()
+	if err != nil {
+		return fmt.Errorf("cannot get best block header: %w", err)
+	}
 
-		s.telemetry.SendMessage(telemetry.NewBlockInterval(
-			&bestHash,
-			best.Number,
-			&finalizedHash,
-			finalised.Number,
-			big.NewInt(int64(s.transactionHandler.TransactionsCount())),
-			big.NewInt(0), // TODO: (ed) determine where to get used_state_cache_size (#1501)
-		))
+	bestHash := best.Hash()
 
-		time.Sleep(s.telemetryInterval)
+	finalised, err := s.blockState.GetHighestFinalisedHeader()
+	if err != nil {
+		return fmt.Errorf("cannot get highehst finalised header: %w", err)
 	}
+
+	finalizedHash := finalised.Hash()
+
+	s.telemetry.SendMessage(telemetry.NewBlockInterval(
+		&bestHash,
+		best.Number,
+		&finalizedHash,
+		finalised.Number,
+		big.NewInt(int64(s.transactionHandler.TransactionsCount())),
+		big.NewInt(0), // TODO: (ed) determine where to get used_state_cache_size (#1501)
+	))
```

Comment:
nit it would be nice to split this in two, one instruction for `blockInterval := ...` and the other for `s.telemetry.SendMessage(blockInterval)`